firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1378,17 +1378,19 @@ class BrowserViewController: UIViewController,
 
         let isPrivate = tabManager.selectedTab?.isPrivate ?? false
         let searchViewModel = SearchViewModel(isPrivate: isPrivate,
-                                              isBottomSearchBar: isBottomSearchBar)
+                                              isBottomSearchBar: isBottomSearchBar,
+                                              profile: profile,
+                                              model: profile.searchEngines,
+                                              tabManager: tabManager)
         let searchController = SearchViewController(profile: profile,
                                                     viewModel: searchViewModel,
-                                                    model: profile.searchEngines,
                                                     tabManager: tabManager)
-        searchController.searchEngines = profile.searchEngines
+        searchViewModel.searchEngines = profile.searchEngines
```

Comment:
not sure why we set it here and pass it through the initializer, but kept as is to minimize issues

Reply:
this seems to be we need to break all this apart 🙈

Reply:
yeah there's a lot of messy code that seems it can be optimized 😅 but not sure if it was done that way for a reason so left them as is for the most part